// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package kv_server;

// Response format of getValuesBinary call in UDFs.
// getValuesBinary will return a Uint8Array with
// a serialized BinaryGetValuesResponse proto.
message BinaryGetValuesResponse {
  // status of the getValuesBinary call
  Status status = 1;
  // key-value pairs
  map<string, Value> kv_pairs = 2;
}

// Value of a key-value pair in the response.
// Can be either status or data.
message Value {
  oneof value {
    Status status = 1;
    bytes data = 2;
  }
}

// Status with code and status message.
message Status {
  // OK is 0, non-zero indicates an error
  int32 code = 1;
  // "ok" on success, error message otherwise
  string message = 2;
}
